components/eventing-controller/pkg/handlers/utils.go

```diff
@@ -297,3 +300,14 @@ func APIRuleGroupVersionResource() schema.GroupVersionResource {
 		Resource: "apirules",
 	}
 }
+
+// NewDeliverPerSubscriptionMetric return a new delivery_per_subscription metrics Collector.
+func NewDeliverPerSubscriptionMetric() *prometheus.CounterVec {
```

Comment:
How about having a new metrics package similar to [this one](https://github.com/kyma-project/kyma/tree/main/components/event-publisher-proxy/pkg/metrics) and make this extensible?

Reply:
Good proposal, I added a prometheus Collector implementation to our controller